internal/serverless/client.go

```diff
@@ -0,0 +1,334 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package serverless
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+const (
+	defaultHostURL = "https://cloud.elastic.co"
+
+	projectsAPI = "/api/v1/serverless/projects"
+)
+
+type Client struct {
+	host   string
+	apiKey string
+}
+
+// ClientOption is functional option modifying Serverless API client.
+type ClientOption func(*Client)
+
+var (
+	ServerlessApiKeyEnvironmentVariable = "EC_API_KEY"
+	ServerlessHostvironmentVariable     = "EC_HOST"
```

Comment:
Nit.
```suggestion
	ElasticCloudApiKeyEnv   = "EC_API_KEY"
	ElasticCloudEndpointEnv = "EC_HOST"
```

Reply:
Updated, and I'll make them private. Currently they are used just here.